core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -424,14 +412,21 @@ private object RTS {
                       if (finalizer eq null) {
                         // No finalizer, so immediately produce the error.
                         curIo = null
-                        result = ExitResult.Failed(error)
+
+                        val ts: List[Throwable] = status.get match {
+                          case Executing(Some(ts), _, _)            => ts
+                          case AsyncRegion(Some(ts), _, _, _, _, _) => ts
+                          case _                                    => Nil
+                        }
+
+                        result = ExitResult.Failed(error, ts)
 
                         // Report the uncaught error to the supervisor:
-                        rts.submit(rts.unsafeRun(unhandled(Errors.UnhandledError(error))))
+                        rts.submit(rts.unsafeRun(unhandled(Errors.UnhandledError(error) :: ts)))
```

Comment:
I am now thinking this is not a good place to report the errors. We could consolidate the logic by putting it into `FiberContext.done`, and ensure that no matter how the fiber terminate, as soon as it's done, all the errors get reported in the same uniform way.